lib/evmone/baseline_execution.cpp

```diff
@@ -51,16 +51,20 @@ inline evmc_status_code check_requirements(const CostTable& cost_table, int64_t&
     auto gas_cost = instr::gas_costs[EVMC_FRONTIER][Op];  // Init assuming const cost.
     if constexpr (!instr::has_const_gas_cost(Op))
     {
-        gas_cost = cost_table[Op];  // If not, load the cost from the table.
+        gas_cost = cost_table[Op];  // If not, load the cost from the current revision cost table.
 
         // Negative cost marks an undefined instruction.
-        // This check must be first to produce correct error code.
-        if (INTX_UNLIKELY(gas_cost < 0))
-            return EVMC_UNDEFINED_INSTRUCTION;
+        // This check must be the first to produce the correct error code.
+        // By definition not possible if defined since the first revision.
+        if constexpr (instr::traits[Op].since == EVMC_FRONTIER)
```

Comment:
I used the opcode trait for the check. Please double check this is correct.

Reply:
so I was pondering this, but opted for `gas_costs` since this is what the assertion above used, and also the entire check here. But if `traits` sounds good to you, I guess it is more natural, :+1:

Reply:
Slightly more readable to me.